app/src/keymap.c

```diff
@@ -793,6 +793,8 @@ static int keymap_handle_set(const char *name, size_t len, settings_read_cb read
             LOG_ERR("Failed to handle keymap layer name from settings (err %d)", err);
             return err;
         }
+
+        zmk_keymap_layer_names[layer][len] = 0;
```

Comment:
We need to be careful just in case someone has changed the `CONFIG_ZMK_KEYMAP_LAYER_NAME_MAX_LEN` value and lowered it since a setting was stored. This should probably be:

```suggestion
        zmk_keymap_layer_names[layer][MIN(len, CONFIG_ZMK_KEYMAP_LAYER_NAME_MAX_LEN - 1)] = 0;
```

To be safe.

Reply:
Maybe using `zmk_keymap_layer_names[layer][err]` would be better? According to the comment in `settings_read_cb`, it represents the number of bytes read.

Reply:
The length of the stored string may be longer than the allocated memory into which we are loading it. So we need to be more conservative. See the previous comment.

Reply:
Yes, but theoretically, `err` shouldn’t be longer than the length passed to `read_cb`, which is already `MIN(len, CONFIG_ZMK_KEYMAP_LAYER_NAME_MAX_LEN - 1)`.

Reply:
Oh indeed! That should be fine.